### ISSUE TITLE
lab of documents

### ISSUE THREAD

**Issue by czt616** [NTMC-Community/MatchZoo#796]
I am using MatchZoo 2.1 to training some models, and I have some questions. When I train the drmm model for document ranking, the label of documents in tutorial is either 0 or 1. I am wondering if I use the cross entropy loss, can I change the value to other value between 0 and 1 to represent different relevance of the document? I added some new documents with a label of 0.75 to wikiqa train set. However, DRMM model still could be trained. Why is this happening?

**Comment by bwanglzu** [comment]
let me elaterate a little bit:

In Matchzoo, everything is *configurable*. Which means model could be used for *classification* (binary) or *ranking* (relevance degree), it's deppend on the *task* type.

To be more concrete, let's see *DRMM* model, line [88](https://github.com/NTMC-Community/MatchZoo/blob/926466ce4bbb29812210ad84c77227456f113c6d/matchzoo/models/drmm.py#L88)

```python
x_out = self._make_output_layer()(flatten_score)
``` 

The output is based on the function called `_make_output_layer`, it's defined in `engine/base_model` line [508](https://github.com/NTMC-Community/MatchZoo/blob/926466ce4bbb29812210ad84c77227456f113c6d/matchzoo/engine/base_model.py#L508)

```python
    def _make_output_layer(self) -> keras.layers.Layer:
        """:return: a correctly shaped keras dense layer for model output."""
        task = self._params['task']
        if isinstance(task, tasks.Classification):
            return keras.layers.Dense(task.num_classes, activation='softmax')
        elif isinstance(task, tasks.Ranking):
            return keras.layers.Dense(1, activation='linear')
        else:
            raise ValueError(f"{task} is not a valid task type."
                             f"Must be in `Ranking` and `Classification`.")
```
